### PR TITLE
fix: saved searches clear all button logic

### PR DESCRIPTION
**Pull request by MounirDhahri**
[ONYX-455]

### Description

This PR fixes an issue where the artist id is getting removed after pressing on the clear all button.

https://github.com/artsy/eigen/assets/11945712/a5bc319b-420b-4b5c-b3e5-3d435d121980

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- fix saved searches clear all button logic - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-455]: https://artsyproduct.atlassian.net/browse/ONYX-455?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (fix saved searches clear all button logic - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f623fa732a11698caf68efcf394e777b107e3316
</p>

File: src/app/Scenes/SavedSearchAlert/SavedSearchStore.tsx [hunk 1]
```diff
@@ -76,7 +76,11 @@ export const savedSearchModel: SavedSearchModel = {
 
   clearAllAttributesAction: action((state) => {
     if (Object.values(state.attributes).length !== 0) {
-      state.attributes = {}
+      state.attributes = {
+        // Keep only selected artist IDs
+        artistIDs: state.attributes.artistIDs,
+        artistID: state.attributes.artistID,
+      }
```
Comment (olerichter00) [inline src/app/Scenes/SavedSearchAlert/SavedSearchStore.tsx:83]: https://lodash.com/docs/4.17.15#pick

```suggestion
      state.attributes = pick(state.attributes, ['artistIDs', 'artistID']);
```

Comment (olerichter00) [reply src/app/Scenes/SavedSearchAlert/SavedSearchStore.tsx:83]: Does the check `if (Object.values(state.attributes).length !== 0)` still make sense here ('artistIDs' or 'artistID' will is present in most cases, right)?

Comment (MounirDhahri) [reply src/app/Scenes/SavedSearchAlert/SavedSearchStore.tsx:83]: true, we can avoid this and use isEqual to set as `dirty`

**Review by olerichter00 — APPROVED** [review]

**Pull request merged**